src/bindings/gpio.rs

```diff
@@ -1,64 +1,133 @@
-use rppal::gpio::Error;
-use rppal::gpio::Gpio;
-use rppal::gpio::OutputPin;
-use std::result::Result::*;
+pub mod gpio_controller;
+use gpio_controller::GpioController;
+use std::sync::atomic::{AtomicBool, Ordering};
+
 /// The enum type representing the avaiable Gpio
-/// .
-/// Each enum value is a Gpio that we can use in our project
+///
+/// Each enum value is a Gpio that you can use
+///
+// If you change the enum variant make sure to assign a integer x that respect 0 <= x < [mem::GPIO_SUPPORTED_NBRS].
+#[derive(Copy, Clone)]
 pub enum GpioPinAvailable {
-    Gpio0,
-    Gpio1,
+    Gpio0 = 0,
+    Gpio1 = 1,
```

Comment:
Is it intentional that `GPIO_SUPPORTED_NBRS = 32` but there is only 2 `GpioPinAvailable` here?

Reply:
It was for the first draft but now everything available is implemented